app/Http/Controllers/HunterController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Exceptions\APIException;
+use App\Hunter;
+use App\Models\Course;
+use Illuminate\Http\Request;
+use Exception;
+use Illuminate\Support\Facades\Validator;
+use App\Http\Controllers\CourseController;
+
+//TODO: post store 寫insert into的方法
+//TODO: get apiIndex 寫select *
+//TODO: get show 寫select by ID
+//TODO: put update 寫update
+//TODO: delete 寫delete 
+class HunterController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+        return Hunter::all()->toArray();
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+
+
+
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //        
+        $validator = Validator::make($request->all(),[
+            'name' => 'required|string|max:20',
+        ]);
+
+        if ($validator->fails()) {
+            throw new APIException('Validation error', 422);
+        }
+
+        $courseForm = [
+            'id' => $request->get('id'),
+            'course_id' => $request->get('course_id'),
+            'name' => $request->get('name'),
+            'description' => $request->get('description') ?? '',
+            'outline' => $request->get('outline') ?? ''
+        ];
+
+        $status = Hunter::create($courseForm);
+
+        return ['success' => $status];
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Hunter  $hunter
+     * @return \Illuminate\Http\Response
+     */
+    public function show($hunter)
+    {
+        //
+        if (Hunter::find($hunter) == null) {
+            throw new APIException('Cannot find this course', 404);
+        } else {
+            return json_encode(Hunter::find($hunter)->toArray());
+        }      
```

Comment:
可以使用 early return 減少層級，方便閱讀
另外 重複的 find 會對執行兩次 query 

```suggestion
        $hunter = Hunter::find($hunter);
        if (empty($hunter)) {
            throw new APIException('Cannot find this course', 404);
        }
        return response()->json($hunter->toArray());
```